EuroPythonBot/registration/pretix_connector.py

```diff
@@ -0,0 +1,126 @@
+            f"{self._pretix_api_url}/orders",
+            params=params,
+        )
+
+        for order_as_json in orders_as_json:
+            order = PretixOrder(**order_as_json)
+            if not order.is_paid:
+                continue
+
+            for position in order.positions:
+                item = self.items_by_id[position.item_id]
+                item_name = item.names_by_locale["en"]
+
+                ticket = Ticket(order=order.id, name=position.attendee_name, type=item_name)
+                self.tickets_by_key[ticket.key] = ticket
+
+    async def _fetch_pretix_items(self) -> None:
+        """Fetch all items from the Pretix API."""
+        items_as_json = await self._fetch_all_pages(f"{self._pretix_api_url}/items")
+
+        for item_as_json in items_as_json:
+            item = PretixItem(**item_as_json)
+            self.items_by_id[item.id] = item
+            for variation in item.variations:
+                self.items_by_id[variation.id] = variation
+
+    async def _fetch_all_pages(self, url: str, params: dict[str, str] | None = None) -> list[dict]:
+        """Fetch all pages from a paginated Pretix API endpoint."""
+        # https://docs.pretix.eu/en/latest/api/fundamentals.html#pagination
+        results = []
+
+        _logger.debug("Fetching all pages from %s (params: %r)", url, params)
+        start = time.perf_counter()
+        async with aiohttp.ClientSession() as session:
+            next_url: str | None = url
+            while next_url is not None:
+                _logger.debug("Fetching %s", url)
+
+                if next_url != url:
+                    params = None  # only send params on initial request
+
+                async with session.get(
+                    next_url, headers=self._http_headers, params=params
+                ) as response:
+                    if response.status != HTTPStatus.OK:
+                        response.raise_for_status()
+
+                    data = await response.json()
+
+                page_results = data["results"]
+                results += page_results
+                next_url = data["next"]
+                _logger.debug("Found %d items", len(page_results))
+
+            _logger.info(
+                "Fetched %d results in %.3f seconds", len(results), time.perf_counter() - start
+            )
+            return results
+
+    def get_ticket(self, *, order: str, name: str) -> Ticket | None:
+        """Get the ticket for a given order ID and name, or None if none was found."""
+        # try different name orders (e.g. family name first vs last)
+        name_parts = name.split(maxsplit=5)
+        for permutation in itertools.permutations(name_parts):
+            possible_name = " ".join(permutation)
+
+            key = generate_ticket_key(order=order, name=possible_name)
+
+            if key in self.tickets_by_key:
+                return self.tickets_by_key[key]
+
+        return None
```

Comment:
We could shrink it further like:
```suggestion
            return self.tickets_by_key.get(key, None)
```

Tested like this:

```Python
>>> a = {"A": 1, "B": 2}
>>> "A" in a
True
>>> a.get("A", None)
1
>>> a.get("C", None)
>>> res = a.get("C", None)
>>> res
>>> type(res)
<class 'NoneType'>
```

Reply:
We want to try all combinations until we find a match, and only return `None` if no combination matches. With your suggestion, we would return `None` if the first combination failed, without trying the other combinations.